python-package/lightgbm/basic.py

```diff
@@ -1068,7 +1082,7 @@ def subset(self, used_indices, params=None):
         """
         if params is None:
             params = self.params
-        ret = Dataset(None, reference=self, feature_name=self.feature_name,
+        ret = Dataset(self.data, reference=self, feature_name=self.feature_name,
```

Comment:
https://github.com/microsoft/LightGBM/pull/1870#discussion_r240071731

Reply:
sorry, I forget the reason of using `None` here, will fix it.